### PR TITLE
refactor: Move proposal function types into a dedicated file

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
Separating internal types from types we import helps make it clear which code needs to be updated regularly.

# Changes
- Move NNS proposal function `defs`, already in their own module, into their own file.
- Add comments.

# Tests
There is no change in functionality or code, so no additional tests are needed.

# Todos

- [ ] Add entry to changelog (if necessary).
  Too minor

### CODE REVIEW

File: rs/proposals/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//! Code to fetch proposals and render them as JSON for human consumption.
```
Comment (mstrasinskis) [inline rs/proposals/src/lib.rs:1]: 🤖 for human consumption :)

File: rs/proposals/src/def.rs [hunk 2]
```diff
@@ -0,0 +1,469 @@
+//! NNS proposal functions.
+//!
+//! If a proposal is passed, the governance canister will typically execute a function call specified in the proposal.
+//! This function call is typically one of the following:
+//! - An update call on some canister, in which case the types are defined in the cod eof the target canister.
```
Comment (mstrasinskis) [inline rs/proposals/src/def.rs:5]: Probably a typo in the: `🐟 end of file`

**Review by mstrasinskis — APPROVED** [review]
Great, thank you!

File: rs/proposals/src/def.rs (hunk 2 above)
Comment (bitdivine) [reply rs/proposals/src/def.rs:5]: Thanks!

**Pull request merged**